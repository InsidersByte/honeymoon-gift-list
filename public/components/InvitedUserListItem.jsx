/* @flow */

import React, { Component } from 'react';
import { Avatar, ListItem, IconButton } from 'material-ui';
import Email from 'material-ui/svg-icons/communication/email';
import Delete from 'material-ui/svg-icons/action/delete';

type PropsType = {
    user: {
        id: number,
        email: string,
    },
    onDelete: Function,
};

export default class InvitedUserListItem extends Component {
    props: PropsType;

    onDelete = () => {
        this.props.onDelete(this.props.user);
    };

    render() {
        const { user: { id, email } } = this.props;

        return (
            <ListItem
                key={id}
                leftAvatar={<Avatar icon={<Email />} />}
                rightIconButton={<IconButton onClick={this.onDelete}><Delete /></IconButton>}
                primaryText={email}
            />
        );
    }
}
