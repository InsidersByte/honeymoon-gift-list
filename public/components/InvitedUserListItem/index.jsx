/* @flow */

import React from 'react';
import { Avatar, ListItem, IconButton } from 'material-ui';
import Email from 'material-ui/svg-icons/communication/email';
import Delete from 'material-ui/svg-icons/action/delete';
import type { UserType } from '../../types';

type PropsType = {
  user: UserType,
  onDelete: Function,
};

const InvitedUserListItem = ({ user, onDelete }: PropsType) => {
  const { id, email } = user;
  const handleDelete = () => onDelete(user);

  return (
    <ListItem
      key={id}
      leftAvatar={<Avatar icon={<Email />} />}
      rightIconButton={<IconButton onClick={handleDelete}><Delete /></IconButton>}
      primaryText={email}
    />
  );
};

export default InvitedUserListItem;
