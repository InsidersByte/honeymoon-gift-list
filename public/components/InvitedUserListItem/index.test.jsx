import React from 'react';
import { shallow, mount } from 'enzyme';
import { IconButton, MuiThemeProvider } from 'material-ui';
import { getMuiTheme } from 'material-ui/styles';
import injectTapEventPlugin from 'react-tap-event-plugin';
import InvitedUserListItem from './';

injectTapEventPlugin();

describe('InvitedUserListItem', () => {
  it('should render correctly', () => {
    const user = { id: 1, email: 'email@email.com', name: 'person' };
    const onDelete = jest.fn();

    const wrapper = shallow(<InvitedUserListItem user={user} onDelete={onDelete} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle delete correctly', () => {
    const user = { id: 1, email: 'email@email.com', name: 'person' };
    const onDelete = jest.fn();

    const wrapper = mount(
      <MuiThemeProvider muiTheme={getMuiTheme()}>
        <InvitedUserListItem user={user} onDelete={onDelete} />
      </MuiThemeProvider>
    );

    expect(wrapper.find(IconButton).length).toBe(1);
    wrapper.find(IconButton).simulate('click');

    expect(onDelete).toBeCalledWith(user);
  });
});
