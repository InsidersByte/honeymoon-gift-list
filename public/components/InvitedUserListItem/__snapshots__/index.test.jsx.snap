// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvitedUserListItem should render correctly 1`] = `
<ListItem
  autoGenerateNestedIndicator={true}
  containerElement="span"
  disableKeyboardFocus={false}
  disabled={false}
  initiallyOpen={false}
  insetChildren={false}
  leftAvatar={
    <Avatar
      icon={<CommunicationEmail />}
      size={40}
    />
  }
  nestedItems={Array []}
  nestedLevel={0}
  onKeyboardFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onNestedListToggle={[Function]}
  onTouchEnd={[Function]}
  onTouchStart={[Function]}
  open={null}
  primaryText="email@email.com"
  primaryTogglesNestedList={false}
  rightIconButton={
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={Object {}}
      onClick={[Function]}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <ActionDelete />
    </IconButton>
  }
  secondaryTextLines={1}
/>
`;
